testScripts/PlayoutObject.js

```diff
@@ -0,0 +1,255 @@
+/* eslint-disable no-console,no-unused-vars */
+
+const fs = require("fs");
+const pathmod = require("path");
+
+const playoutFormats =
+      {
+        "dash-clear": {
+          "drm": null,
+          "protocol": {
+            "min_buffer_length": 2,
+            "type": "ProtoDash"
+          }
+        },
+        "hls-clear": {
+          "drm": null,
+          "protocol": {
+            "type": "ProtoHls"
+          }
+        }
+      };
+
+// User options
+const streamIdAudio = "audio_2ch_eng";
+const streamIdVideo = "video_1920x1080_und";
+const streamIdCaptions = "captions_eng";
+
+const contentEncryption = "none"; // options "cgck" and "none"
+const reps = {};
+
+reps[streamIdAudio] =
+  {
+    "stereo@128000": {
+      "bit_rate": 128000,
+      "media_struct_stream_key": streamIdAudio,
+      "type": "RepAudio"
+    }
+  };
+
+reps[streamIdVideo] =
+  {
+    "1080@6500000": {
+      "bit_rate": 6500000,
+      "crf": 0,
+      "height": 1080,
+      "media_struct_stream_key": streamIdVideo,
+      "type": "RepVideo",
+      "width": 1920
+    },
+    "360@400000": {
+      "bit_rate": 400000,
+      "crf": 0,
+      "height": 360,
+      "media_struct_stream_key": streamIdVideo,
+      "type": "RepVideo",
+      "width": 640
+    },
+    "432@1100000": {
+      "bit_rate": 1100000,
+      "crf": 0,
+      "height": 432,
+      "media_struct_stream_key": streamIdVideo,
+      "type": "RepVideo",
+      "width": 768
+    },
+    "540@2000000": {
+      "bit_rate": 2000000,
+      "crf": 0,
+      "height": 540,
+      "media_struct_stream_key": streamIdVideo,
+      "type": "RepVideo",
+      "width": 960
+    },
+    "720@4500000": {
+      "bit_rate": 4500000,
+      "crf": 0,
+      "height": 720,
+      "media_struct_stream_key": streamIdVideo,
+      "type": "RepVideo",
+      "width": 1280
+    }
+  };
+
+reps[streamIdCaptions] =
+  {
+    "vtt": {
+      "bit_rate": 100,
+      "media_struct_stream_key": streamIdCaptions,
+      "type": "RepCaptions"
+    }
+  };
+
+const ReadDir = (path) => {
+
+  var pathlist = [];
+  const dir = fs.readdirSync(path, {withFileTypes: true});
+
+  console.log("Read dir: " + path);
+  dir.forEach(item => {
+    const itemPath = pathmod.join(path, item.name);
+    console.log("- path: " + itemPath);
+    if(item.isFile()) {
+      pathlist.push(itemPath);
+    } else if (item.isDirectory()) {
+      dirpath = ReadDir(itemPath);
+      console.log("Subdir dirpath: " + JSON.stringify(dirpath));
+      pathlist = pathlist.concat(dirpath);
+    }
+  });
+  return pathlist;
+};
+
+class PlayoutObject {
+
+  constructor(elvclient) {
+    this.elvclient = elvclient;
+  }
+
+  async uploadDir(dir) {
+
+    const pathlist = await ReadDir(dir);
+    var mediaStructFile;
+
+    try {
+
+      for (var i = 0, len = pathlist.length; i < len; i++) {
+	var singleFilePath = pathlist[i];
+
+	console.log("Reading file: " + singleFilePath + " ...");
+	var fileContents = fs.readFileSync(singleFilePath);
```

Comment:
How large are these files? We shouldn't read them into memory, but upload them using a stream/reader.